### PR TITLE
Check elements after initial setup

### PR DESCRIPTION
**Pull request by autarc**
The current implementation ignores asynchronous added elements and changes, so the `registry.check()` will only consider the matches from the initial setup. With the addition of querying for new or removed entries its now solved.

In addition the issue of calling the library after the `DOMContentLoaded` and `load` event were already triggered got fixed. Furthermore an initial check will be executed to ensure consistent behavior from the start without requiring an user interaction.

### CODE REVIEW

File: src/in-view.js [hunk 1]
```diff
@@ -17,40 +17,46 @@ const inView = () => {
...
     */
     const check = throttle(() => {
         selectors.history.forEach(selector => {
-            selectors[selector].check();
+            const registry = selectors[selector];
+            if (options.async) {
+                let elements = [].slice.call(document.querySelectorAll(selector));
+                if (elements.length !== registry.elements.length) {
+                    registry.elements = elements;
+                }
```
Comment (sahandnayebaziz) [inline src/in-view.js:40]: What does this check do? Is it meant to save time? To me, it looks like it's here to save an operation if the number of elements turns out to be the same number of elements we already have, but what if we have the same number of elements but there are all new?

File: src/in-view.js [hunk 2]
```diff
@@ -17,40 +17,46 @@ const inView = () => {
...
     /**
     * If supported, use MutationObserver to watch the
     * DOM and run checks on mutation.
     */
     if (window.MutationObserver) {
-        addEventListener('DOMContentLoaded', () => {
+        isReady(() => {
             new MutationObserver(check)
-                .observe(document.body, { attributes: true, childList: true, subtree: true });
+                .observe(document, { attributes: true, childList: true, subtree: true });
```
Comment (sahandnayebaziz) [inline src/in-view.js:59]: Is removing the body here significant? It seems unrelated to checking for new elements that weren't there at first call to inView.

Comment (autarc) [reply src/in-view.js:59]: It depends on the time when the library is loaded and used. If the DOM is not ready upfront as creating a browser extension its better to specify it as general as possible.

File: src/in-view.js (hunk 1 above)
Comment (autarc) [reply src/in-view.js:40]: Its meant as a quick dirty check for the matches. Could also apply a deeper comparison, bit as the call will already be throttled its probably less important.

Comment (sahandnayebaziz) [reply src/in-view.js:40]: To me, it seems like unless we do a deep comparison, this isn't helpful because we could have a situation where...

```javascript
elements.length == 1 

// and

registry.elements.length == 1

// but the elements are actually different, and since we only check length,
// we don't set them to the registry.
```
